pkg/store/bucket.go

```diff
@@ -2563,7 +2629,19 @@ func matchersToPostingGroups(ctx context.Context, lvalsFn func(name string) ([]s
 			if !mergedPG.addAll && len(mergedPG.addKeys) == 0 {
 				return nil, nil
 			}
-		}
+			matchers = append(matchers, val)
+		}
+		// Set and sort matchers to be used when picking up posting fetch strategy.
+		mergedPG.matchers = matchers
+		slices.SortFunc(mergedPG.matchers, func(a, b *labels.Matcher) bool {
```

Comment:
Same here, we could use `sortedMatchers`

Reply:
Updated